CUDA Error running the Translation example due to embeddings

### System Info

Hello Team, 

I am trying to run the translation example in examples/pytorch/translation/run_translation.py in a distributed manner through accelerate as follows.

```bash
accelerate launch --config_file default_config.yaml run_translation.py \
    --model_name_or_path Helsinki-NLP/opus-mt-en-ro \
    --do_train \
    --do_eval \
    --source_lang en \
    --target_lang ro \
    --dataset_name wmt16 \
    --dataset_config_name ro-en \
    --output_dir /tmp/tst-translation \
    --per_device_train_batch_size=4 \
    --per_device_eval_batch_size=4 \
    --overwrite_output_dir \
    --predict_with_generate \
    --pad_to_max_length True \
    --report_to none
```

**Accelerator Config**
```bash
compute_environment: LOCAL_MACHINE
debug: false
distributed_type: MULTI_GPU
downcast_bf16: 'no'
gpu_ids: 0,1
machine_rank: 0
main_training_function: main
mixed_precision: fp16
num_machines: 1
num_processes: 2
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false
```

But I see the following CUDA error. Could you please help me to understand what changes I need to make. I have run other examples in the summarization and the language-modeling folder in a similar manner successfully.

**Python venv**
```
transformers==4.35.2
accelerate==0.25.0
datasets==2.15.0
```

**Error Logs**
```
../aten/src/ATen/native/cuda/Indexing.cu:1292: indexSelectLargeIndex: block: [421,0,0], thread: [60,0,0] Assertion `srcIndex < srcSelectDimSize` failed.
../aten/src/ATen/native/cuda/Indexing.cu:1292: indexSelectLargeIndex: block: [421,0,0], thread: [61,0,0] Assertion `srcIndex < srcSelectDimSize` failed.
../aten/src/ATen/native/cuda/Indexing.cu:1292: indexSelectLargeIndex: block: [421,0,0], thread: [62,0,0] Assertion `srcIndex < srcSelectDimSize` failed.
../aten/src/ATen/native/cuda/Indexing.cu:1292: indexSelectLargeIndex: block: [421,0,0], thread: [63,0,0] Assertion `srcIndex < srcSelectDimSize` failed.
Traceback (most recent call last):
  File "run_translation.py", line 699, in <module>
    main()
  File "run_translation.py", line 614, in main
    train_result = trainer.train(resume_from_checkpoint=checkpoint)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/transformers/trainer.py", line 1555, in train
    return inner_training_loop(
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/transformers/trainer.py", line 1860, in _inner_training_loop
    tr_loss_step = self.training_step(model, inputs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/transformers/trainer.py", line 2725, in training_step
    loss = self.compute_loss(model, inputs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/transformers/trainer.py", line 2748, in compute_loss
    outputs = model(**inputs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 1519, in forward
    else self._run_ddp_forward(*inputs, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 1355, in _run_ddp_forward
    return self.module(*inputs, **kwargs)  # type: ignore[index]
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/accelerate/utils/operations.py", line 680, in forward
    return model_forward(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/accelerate/utils/operations.py", line 668, in __call__
    return convert_to_fp32(self.model_forward(*args, **kwargs))
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/amp/autocast_mode.py", line 16, in decorate_autocast
    return func(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/transformers/models/marian/modeling_marian.py", line 1402, in forward
    outputs = self.model(
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/transformers/models/marian/modeling_marian.py", line 1185, in forward
    encoder_outputs = self.encoder(
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/transformers/models/marian/modeling_marian.py", line 739, in forward
    hidden_states = inputs_embeds + embed_pos
RuntimeError: CUDA error: device-side assert triggered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

  0%|                                                                                                             | 0/228870 [00:03<?, ?it/s]
terminate called after throwing an instance of 'c10::Error'
  what():  CUDA error: device-side assert triggered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

Exception raised from c10_cuda_check_implementation at ../c10/cuda/CUDAException.cpp:44 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7f7f442b5617 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x64 (0x7f7f4427098d in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #2: c10::cuda::c10_cuda_check_implementation(int, char const*, char const*, int, bool) + 0x118 (0x7f7f44371128 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10_cuda.so)
frame #3: <unknown function> + 0x16e76 (0x7f7f44339e76 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10_cuda.so)
frame #4: <unknown function> + 0x19bad (0x7f7f4433cbad in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10_cuda.so)
frame #5: <unknown function> + 0x19fcd (0x7f7f4433cfcd in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10_cuda.so)
frame #6: <unknown function> + 0x510c56 (0x7f7f448dcc56 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
frame #7: <unknown function> + 0x55ca7 (0x7f7f4429aca7 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #8: c10::TensorImpl::~TensorImpl() + 0x1e3 (0x7f7f44292cb3 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #9: c10::TensorImpl::~TensorImpl() + 0x9 (0x7f7f44292e49 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #10: <unknown function> + 0x7c1718 (0x7f7f44b8d718 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
frame #11: THPVariable_subclass_dealloc(_object*) + 0x325 (0x7f7f44b8dac5 in /home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
frame #12: /home/anindya/starcoder-tune/bin/python3() [0x5aced3]
frame #13: /home/anindya/starcoder-tune/bin/python3() [0x5b0174]
frame #14: /home/anindya/starcoder-tune/bin/python3() [0x5f7cdd]
frame #15: /home/anindya/starcoder-tune/bin/python3() [0x5b02f0]
frame #16: /home/anindya/starcoder-tune/bin/python3() [0x5835c2]
frame #17: /home/anindya/starcoder-tune/bin/python3() [0x4c518f]
frame #18: _PyGC_CollectNoFail + 0x2f (0x66721f in /home/anindya/starcoder-tune/bin/python3)
frame #19: PyImport_Cleanup + 0x244 (0x67a634 in /home/anindya/starcoder-tune/bin/python3)
frame #20: Py_FinalizeEx + 0x7f (0x67423f in /home/anindya/starcoder-tune/bin/python3)
frame #21: Py_RunMain + 0x32d (0x6b418d in /home/anindya/starcoder-tune/bin/python3)
frame #22: Py_BytesMain + 0x2d (0x6b43fd in /home/anindya/starcoder-tune/bin/python3)
frame #23: __libc_start_main + 0xf3 (0x7f7f59353083 in /lib/x86_64-linux-gnu/libc.so.6)
frame #24: _start + 0x2e (0x5da67e in /home/anindya/starcoder-tune/bin/python3)

[2023-12-18 06:41:41,495] torch.distributed.elastic.multiprocessing.api: [ERROR] failed (exitcode: -6) local_rank: 0 (pid: 369953) of binary: /home/anindya/starcoder-tune/bin/python3
Traceback (most recent call last):
  File "/home/anindya/starcoder-tune/bin/accelerate", line 8, in <module>
    sys.exit(main())
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/accelerate/commands/accelerate_cli.py", line 47, in main
    args.func(args)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/accelerate/commands/launch.py", line 1008, in launch_command
    multi_gpu_launcher(args)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/accelerate/commands/launch.py", line 666, in multi_gpu_launcher
    distrib_run.run(args)
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/distributed/run.py", line 797, in run
    elastic_launch(
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/home/anindya/starcoder-tune/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 264, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
run_translation.py FAILED
------------------------------------------------------------
```

### Who can help?

@patil-suraj @pacman100 @ArthurZucker 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

STEP 1: Create a basic Accelerator config `default_config.yaml` file with 2 GPUs m/c as below.

```bash
compute_environment: LOCAL_MACHINE
debug: false
distributed_type: MULTI_GPU
downcast_bf16: 'no'
gpu_ids: 0,1
machine_rank: 0
main_training_function: main
mixed_precision: fp16
num_machines: 1
num_processes: 2
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false
```

STEP 2: Run the translation example.

```bash
accelerate launch --config_file default_config.yaml run_translation.py \
    --model_name_or_path Helsinki-NLP/opus-mt-en-ro \
    --do_train \
    --do_eval \
    --source_lang en \
    --target_lang ro \
    --dataset_name wmt16 \
    --dataset_config_name ro-en \
    --output_dir /tmp/tst-translation \
    --per_device_train_batch_size=4 \
    --per_device_eval_batch_size=4 \
    --overwrite_output_dir \
    --predict_with_generate \
    --pad_to_max_length True \
    --report_to none
```



### Expected behavior

The example should complete without any error.

--- a/examples/pytorch/translation/run_translation.py
+++ b/examples/pytorch/translation/run_translation.py
@@ -469,6 +469,19 @@ def main():
     source_lang = data_args.source_lang.split("_")[0]
     target_lang = data_args.target_lang.split("_")[0]
 
+    # Check the whether the source target length fits in the model, if it has absolute positional embeddings
+    if (
+        hasattr(model.config, "max_position_embeddings")
+        and not hasattr(model.config, "relative_attention_max_distance")
+        and model.config.max_position_embeddings < data_args.max_source_length
+    ):
+        raise ValueError(
+            f"`--max_source_length` is set to {data_args.max_source_length}, but the model only has"
+            f" {model.config.max_position_embeddings} position encodings. Consider either reducing"
+            f" `--max_source_length` to {model.config.max_position_embeddings} or using a model with larger position "
+            "embeddings"
+        )
+
     # Temporarily set max_target_length for training.
     max_target_length = data_args.max_target_length
     padding = "max_length" if data_args.pad_to_max_length else False

